dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -232,61 +246,61 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
         &hKey,
         nullptr /* lpdwDisposition */));
 
-    // Retrieve DisplayName and IconUri
-    // - DisplayName: Retrieve from Shell. If not specified, fall back to filename.
-    // - Icon: Retrieve from Shell. If it's not the case or the file extension is unsupported, then throw.
-    winrt::com_ptr<IPropertyStore> propertyStore;
-    wil::unique_hwnd hWindow{ GetConsoleWindow() };
-    THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
-
     // Retrieve the display name
     std::wstring displayName{};
 
-    wil::unique_prop_variant propVariantDisplayName;
-    // Do not throw in case of failure. We can use the filepath approach below as fallback to set a DisplayName.
-    LOG_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
-    if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr)
-    {
-        displayName = propVariantDisplayName.pwszVal;
-    }
+    // Retrieve the icon
+    std::wstring iconFilePath{};
 
-    if (displayName.length() == 0)
-    {
-        THROW_IF_FAILED(wil::GetModuleFileNameExW(GetCurrentProcess(), nullptr, displayName));
+    wil::unique_hwnd hWindow{ GetConsoleWindow() };
 
-        size_t lastBackslashPosition{ displayName.rfind(L"\\") };
-        THROW_HR_IF(E_UNEXPECTED, lastBackslashPosition == std::wstring::npos);
-        displayName = displayName.substr(lastBackslashPosition + 1); // One after the delimiter
+    if (hWindow)
+    {
+        // Retrieve DisplayName and IconUri
+        // - DisplayName: Retrieve from Shell. If not specified, fall back to filename.
+        // - Icon: Retrieve from Shell. If it's not the case or the file extension is unsupported, then throw.
+        winrt::com_ptr<IPropertyStore> propertyStore;
+        THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
+
+        wil::unique_prop_variant propVariantDisplayName;
+        // Do not throw in case of failure. We can use the filepath approach below as fallback to set a DisplayName.
+        LOG_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
+        if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr)
+        {
+            displayName = propVariantDisplayName.pwszVal;
+        }
+        else if (displayName.empty())
+        {
+            displayName = SetDisplayNameBasedOnProcessName();
```

Comment:
Overcomplicaed. Set this in the beginning before the if clause. This is the baseline.